tcms/rpc/api/testexecution.py

```diff
@@ -149,33 +149,44 @@ def update(execution_id, values, **kwargs):
         :raises: PermissionDenied if missing *testruns.change_testexecution* permission
     """
 
-    tcr = TestExecution.objects.get(pk=execution_id)
+    execution = TestExecution.objects.get(pk=execution_id)
     form = UpdateExecutionForm(values)
 
     if form.is_valid():
         if form.cleaned_data['build']:
-            tcr.build = form.cleaned_data['build']
+            execution.build = form.cleaned_data['build']
 
         if form.cleaned_data['assignee']:
-            tcr.assignee = form.cleaned_data['assignee']
+            execution.assignee = form.cleaned_data['assignee']
 
         if form.cleaned_data['status']:
-            tcr.status = form.cleaned_data['status']
+            execution.status = form.cleaned_data['status']
             request = kwargs.get(REQUEST_KEY)
-            tcr.tested_by = request.user
+            execution.tested_by = request.user
 
         if form.cleaned_data['sortkey'] is not None:
-            tcr.sortkey = form.cleaned_data['sortkey']
+            execution.sortkey = form.cleaned_data['sortkey']
 
         if form.cleaned_data['tested_by']:
-            tcr.tested_by = form.cleaned_data['tested_by']
+            execution.tested_by = form.cleaned_data['tested_by']
+
+        case_text_version = form.cleaned_data['case_text_version']
+        if case_text_version:
+            update_case_text_version(execution, case_text_version)
 
-        tcr.save()
+        execution.save()
 
     else:
         raise ValueError(form_errors_to_list(form))
 
-    return tcr.serialize()
+    return execution.serialize()
+
+
+def update_case_text_version(execution, case_text_version):
```

Comment:
I don't mind this being a separate method for better abstraction and less if nesting, but it should be a private one:
```python
def _update_case_text_version(execution, case_text_version):
```